File: client.old/GhostClient/Core/Network/Packets/ItemDeactivatedPacket.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using Ghost.Sprites.Effects;
using GhostClient.Core;

namespace Ghost.Core.Network.Packets
{
    public class ItemDeactivatedPacket : Packet
    {
        public ItemDeactivatedPacket(Stream stream) : base(stream)
        {
        }

        public ItemDeactivatedPacket(Stream stream, bool autoClose) : base(stream, autoClose)
        {
        }

        public ItemDeactivatedPacket(byte[] data) : base(data)
        {
        }

        protected override void OnHandlePacket()
        {
            short id = Consume(2).ShortValue;
            short owner = Consume(2).ShortValue;

            if (id == 12)
            {
                var e = GameHandler.Game.FindEntity(owner) as NetworkPlayer;
                if (e != null)
                {
                    e.Orbits[0].End();
                }
            }
            Console.WriteLine("Item " + id + " was deactivated!");
        }
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/ItemDeactivatedPacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.core.sprites.NetworkPlayer
import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient

class ItemDeactivatedPacket : Packet<PlayerClient>() {

    override fun handle() {
        val id = consume(2).asShort()
        val owner = consume(2).asShort()

        if (id == 12.toShort()) {
            val e = client.game.findEntity(owner)

            if (e is NetworkPlayer) {
                e.orbits[0].end()
            }
        }

        System.out.println("Item $id was deactivated for $owner!");
    }
}
